src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/scanneddocs/ManageScannedDocsContestedAboutToSubmitHandler.java

```diff
@@ -64,18 +68,27 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
         log.info("Received request to manage scanned documents for Case ID : {}", callbackRequest.getCaseDetails().getId());
         FinremCaseData caseData = callbackRequest.getCaseDetails().getData();
         List<String> warnings = new ArrayList<>();
-
         getValidatedResponse(caseData, warnings);
-        List<UploadCaseDocumentCollection> manageScannedDocumentCollection = caseData.getManageScannedDocumentCollection();
+
+        List<UploadCaseDocumentCollection> manageScannedDocumentCollection =
+            caseData.getManageScannedDocumentCollection().stream()
+                .filter(msdc -> YesOrNo.YES == msdc.getManageScannedDocument().getSelectForUpdate())
+                .map(ManageScannedDocumentCollection::toUploadCaseDocumentCollection)
+                .collect(Collectors.toCollection(Lists::newArrayList));
 
         updateFileNames(manageScannedDocumentCollection);
         addDefaultsToAdministrativeDocuments(manageScannedDocumentCollection);
 
+        List<String> processedScannedDocumentIds = manageScannedDocumentCollection.stream()
+            .map(UploadCaseDocumentCollection::getId)
+            .collect(Collectors.toCollection(Lists::newArrayList));
```

Comment:
maybe just collect(Collectors.toList())

Reply:
Unfortunately the DocumentHandler code removes items from the list so it needs to be mutable

Reply:
toList() should return a mutable list https://www.baeldung.com/java-stream-to-list-collecting

Reply:
The Javadoc and code clearly show otherwise.
```
    default List<T> toList() {
        return (List<T>) Collections.unmodifiableList(new ArrayList<>(Arrays.asList(this.toArray())));
    }
```

Reply:
no it doesn't, I don't know where you got this code snippet from but looking into the Collectors class here I have this which confirms what baeldung is saying. Happy to have a call to clarify.      

/**
     * Returns a {@code Collector} that accumulates the input elements into a
     * new {@code List}. There are no guarantees on the type, mutability,
     * serializability, or thread-safety of the {@code List} returned; if more
     * control over the returned {@code List} is required, use {@link #toCollection(Supplier)}.
     *
     * @param <T> the type of the input elements
     * @return a {@code Collector} which collects all the input elements into a
     * {@code List}, in encounter order
     */
    public static <T>
    Collector<T, ?, List<T>> toList() {
        return new CollectorImpl<>(ArrayList::new, List::add,
                                   (left, right) -> { left.addAll(right); return left; },
                                   CH_ID);
    }

Reply:
Yep my mistake. Updated